QEfficient/transformers/cache_utils.py

```diff
@@ -78,3 +78,57 @@ def update(
             v_out = torch.where(invalid_mask.unsqueeze(-1), torch.tensor(0.0, dtype=torch.float32), v_out)
 
         return k_out, v_out
+
+    def update3D(
```

Comment:
nit: same goes here, can we check shape of k/v and update 3D in the same update api()?

Reply:
Okay sure, will add CB mode too in this

Reply:
I think we should be able to do this. @quic-akuruvil This add lot of duplicate code.

Reply:
Combining it with the update api require again restructuring the code, which introduces some effort. Can I add it in another PR?